song_rater/views.py

```diff
@@ -28,12 +28,15 @@ def song_detail(request, id):
             rating = form.save(commit=False)
             rating.song_id = id
             rating.save()
-            return redirect("song_rater:song_list")
+            return redirect("song_rater:song_list",)
     else:
         form = RatingForm()
 
-    context = {"details": get_object_or_404(Song, pk=id),
+    song = get_object_or_404(Song, pk=id)
+    ratings_list = song.rating_set.all()
+    context = {"details": song,
```

Comment:
I think you can "inline" the `song` variable.  e.g
```
context = {
    "details":  get_object_or_404(Song, pk=id)
    ...
```

Reply:
I now see you can't. You need it to obtain the `rating_list`.  I like this code the way it is

Reply:
@WangRemy I also like the way you changed indentation.